openhantek/src/glscope.cpp

```diff
@@ -164,9 +167,9 @@ void GlScope::initializeGL() {
 
     qDebug() << "compile shaders";
     // Compile vertex shader
-    bool usesOpenGL = QSurfaceFormat::defaultFormat().renderableType()==QSurfaceFormat::OpenGL;
-    if (!program->addShaderFromSourceCode(QOpenGLShader::Vertex, usesOpenGL ? vshaderDesktop : vshaderES) ||
-        !program->addShaderFromSourceCode(QOpenGLShader::Fragment, usesOpenGL ? fshaderDesktop : fshaderES)) {
+    bool usesVersion_1_5 = QGLFormat::openGLVersionFlags().testFlag(QGLFormat::OpenGL_Version_3_2);
```

Comment:
In the above `fixOpenGLversion` function we explicitly set the `renderType()` to QSurfaceFormat::OpenGL or QSurfaceFormat::OpenGLES. Therefore we can use that information. If you test again for the flag here, the user override to force OpenGLES is different to the shader you select here.

Reply:
Use the method `context()->isOpenGLES()`. That should do it.